### ISSUE TITLE
Cannot read property sourceAndMap of undefined

### ISSUE THREAD

**Issue by phaistonian** [webpack-contrib/terser-webpack-plugin#242]
`Cannot read property 'sourceAndMap' of undefined`

This has started since 3.0.0.

`terser-webpack-plugin/dist/index.js:176:47)`

ps: sorry for not proving much more context.

**Comment by BiscuiTech** [comment]
This has happened to me as well. I'm using it in a next.js app with the following config:
```javascript
require('dotenv').config();
const withSourceMaps = require('@zeit/next-source-maps')();
const TerserPlugin = require('terser-webpack-plugin');

module.exports = withSourceMaps({
  webpack: (config, options) => {
    // Fixes npm packages that depend on `fs` module
    config.node = {
      fs: 'empty',
    };

    if (!options.isServer) {
      config.resolve.alias['@sentry/node'] = '@sentry/browser';
      config.resolve.alias['react-dom$'] = 'react-dom/profiling';
      config.resolve.alias['scheduler/tracing'] = 'scheduler/tracing-profiling';
      config.optimization.minimizer = [
        new TerserPlugin({
          terserOptions: {
            keep_classnames: false,
            keep_fnames: false,
          },
        }),
      ];
    }
    return config;
  },
  experimental: {
    modern: true,
    async rewrites() {
      return [{ source: '/sitemap.xml', destination: '/api/sitemap.xml' }];
    },
    catchAllRouting: true,
  },
  env: {
    PRISMIC_API_TOKEN: process.env.PRISMIC_API_TOKEN,
    PRISMIC_REPOSITORY_NAME: process.env.PRISMIC_REPOSITORY_NAME,
    NOW_URL: process.env.NOW_URL,
  },
});
```

**Comment by evilebottnawi** [comment]
Somebody can create minimum reproducible test repo?

**Comment by evilebottnawi** [comment]
In fact, I know a potential fix, but I do not want to apply it because I would like to see what is the real reason, in order to prevent such regressions in the future

**Comment by evilebottnawi** [comment]
It is happens, when you use plugin/tool for run multi compilation in parallel, WIP

**Comment by BiscuiTech** [comment]
I can confirm the latest version fixes the issue - only if I switch experimental.modern to false.

**Comment by evilebottnawi** [comment]
@BiscuiTech What is `experimental.modern`?

**Comment by Finesse** [comment]
Still an issue in 3.0.1:

```
TypeError: Cannot read property 'source' of undefined
    at TerserPlugin.taskGenerator (~/node_modules/terser-webpack-plugin/dist/index.js:188:27)
    at taskGenerator.next (<anonymous>)
    at ~/node_modules/terser-webpack-plugin/dist/index.js:415:54
    at ~/node_modules/terser-webpack-plugin/node_modules/p-try/index.js:4:10
    at new Promise (<anonymous>)
    at pTry (~/node_modules/terser-webpack-plugin/node_modules/p-try/index.js:3:37)
    at run (~/node_modules/terser-webpack-plugin/node_modules/p-limit/index.js:23:18)
    at next (~/node_modules/terser-webpack-plugin/node_modules/p-limit/index.js:16:17
```

**Comment by Finesse** [comment]
@evilebottnawi Still an issue in v3.0.1 with Webpack v4.43.0:

```
TypeError: Cannot read property 'source' of undefined
    at TerserPlugin.taskGenerator (~/node_modules/terser-webpack-plugin/dist/index.js:188:27)
    at taskGenerator.next (<anonymous>)
    at ~/node_modules/terser-webpack-plugin/dist/index.js:415:54
    at ~/node_modules/terser-webpack-plugin/node_modules/p-try/index.js:4:10
    at new Promise (<anonymous>)
    at pTry (~/node_modules/terser-webpack-plugin/node_modules/p-try/index.js:3:37)
    at run (~/node_modules/terser-webpack-plugin/node_modules/p-limit/index.js:23:18)
    at next (~/node_modules/terser-webpack-plugin/node_modules/p-limit/index.js:16:17
```

<details>
<summary>My setup (a part of)</summary>

```js
// package.json
{
  "scripts": {
    "build": "webpack --mode=production --progress",
  },
  "engines": {
    "node": ">=12"
  },
  "dependencies": {
    "@babel/core": "^7.9.6",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/plugin-transform-runtime": "^7.9.6",
    "@babel/preset-env": "^7.9.6",
    "@babel/preset-react": "^7.9.4",
    "@babel/preset-typescript": "^7.9.0",
    "@babel/runtime": "^7.9.2",
    "autoprefixer": "^9.7.6",
    "babel-loader": "^8.1.0",
    "core-js": "^3.6.5",
    "css-loader": "^3.5.3",
    "csso-webpack-plugin": "^2.0.0-beta.1",
    "es6-error": "^4.0.2",
    "glob": "^7.1.6",
    "ignore-emit-webpack-plugin": "^2.0.2",
    "mini-css-extract-plugin": "^0.9.0",
    "minimatch": "^3.0.4",
    "nanoid": "^3.1.7",
    "postcss-csso": "^4.0.0",
    "postcss-loader": "^3.0.0",
    "react-friendly-input": "^0.1.2",
    "react-popper": "^1.3.7",
    "redux": "^4.0.5",
    "style-loader": "^1.2.1",
    "stylus": "^0.54.7",
    "stylus-loader": "^3.0.2",
    "terser-webpack-plugin": "^3.0.1",
    "translit": "^0.1.0",
    "translit-russian": "^0.0.2",
    "url-loader": "^4.1.0",
    "webfonts-loader": "^5.2.2",
    "webpack": "^4.43.0",
    "webpack-cli": "^3.3.11"
  },
  "devDependencies": {
    "@types/react": "^16.9.35",
    "@types/react-dom": "^16.9.8",
    "@types/react-sortable-hoc": "^0.7.1",
    "@types/resize-observer-browser": "^0.1.3",
    "sweetalert2": "~7.1.2",
    "typescript": "^3.9.2"
  }
}
```

```js
// webpack.config.js
const TerserPlugin = require('terser-webpack-plugin');
const CssoWebpackPlugin = require('csso-webpack-plugin').default;

module.exports = {
    externals: {
        jquery: 'jQuery',
        react: 'React',
        'react-dom': 'ReactDOM',
        tinycolor2: 'tinycolor',
        html2canvas: 'html2canvas',
        'react-sortable-hoc': 'SortableHOC',
        'popper.js': 'Popper',
        'sweetalert2': 'swal'
    },
    module: {
        rules: [
            {
                test: /\.(js|jsx|ts|tsx)$/,
                exclude: /node_modules/,
                loader: {
                    loader: 'babel-loader',
                    options: {
                        presets: [
                            ['@babel/preset-env', {
                                targets: ['cover 94%'],
                                useBuiltIns: 'usage',
                                corejs: '3.0',
                                bugfixes: true,
                            }],
                            '@babel/preset-react',
                            '@babel/preset-typescript'
                        ],
                        plugins: [
                            ['@babel/plugin-transform-runtime', {
                                regenerator: false
                            }],
                            '@babel/plugin-proposal-class-properties'
                        ]
                    }
                }
            }
        ]
    },
    output: {
        path: path.resolve(__dirname, '../../assets'),
        filename: '[name].js'
    },
    optimization: {
        minimizer: [
            new TerserPlugin({
                extractComments: false,
                terserOptions: {
                    mangle: {
                        eval: true
                    },
                    output: {
                        max_line_len: 4095
                    },
                    safari10: true
                }
            }),
        ]
    },
    plugins: [
        new CssoWebpackPlugin()
    ],
    stats: {
        children: false
    }
};
```

</details>

**Comment by evilebottnawi** [comment]
@Finesse Can you open a new issue with reproducible test repo?

**Comment by KeionneDerousselle** [comment]
I am also experiencing the same issue as @Finesse. I have turned on the parallel flag as well and it didn't help.

**Comment by evilebottnawi** [comment]
Please open a new issue with reproducible test repo

**Comment by evilebottnawi** [comment]
Please update `terser-webpack-plugin` to `3.0.5` (https://github.com/webpack-contrib/terser-webpack-plugin/releases/tag/v3.0.5), it should help with `Cannot read property 'source' of undefined` and `Cannot read property 'sourceAndMap' of undefined`
